src/extensions/mountainview/MountainView/ViewContainer.tsx

```diff
@@ -1,46 +1,117 @@
-import React, { FunctionComponent, useEffect, useState } from 'react';
-import sizeMe, { SizeMeProps } from 'react-sizeme';
+import React, { FunctionComponent, useCallback, useEffect, useState } from 'react';
 import { View } from './MountainView';
 import ViewContainerTabBar from './ViewContainerTabBar';
 
 
 type Props = {
     views: View[]
     onViewClosed: (v: View) => void
-    width?: number
+    onSetViewArea: (v: View, area: 'north' | 'south') => void
+    width: number
+    height: number
 }
 
-const tabBarHeight = 56
+const tabBarHeight = 60
 
-const ViewContainer: FunctionComponent<Props & SizeMeProps> = ({ children, views, onViewClosed, size, width }) => {
-    const [currentView, setCurrentView] = useState<View | null>(null)
+const ViewContainer: FunctionComponent<Props> = ({ children, views, onViewClosed, onSetViewArea, width, height }) => {
+    const [currentNorthView, setCurrentNorthView] = useState<View | null>(null)
+    const [currentSouthView, setCurrentSouthView] = useState<View | null>(null)
+    const [activeArea, setActiveArea] = useState<'north' | 'south'>('north')
     useEffect(() => {
         views.forEach(v => {
+            if (!v.area) v.area = activeArea
             if (v.activate) {
                 v.activate = false
-                setCurrentView(v)
+                if (v.area === 'north') {
+                    setCurrentNorthView(v)
+                    setActiveArea('north')
+                }
+                else if (v.area === 'south') {
+                    setCurrentSouthView(v)
+                    setActiveArea('south')
+                }
             }
         })
-    }, [views])
+    }, [views, activeArea])
+
+    const H0 = (height - tabBarHeight * 2) / 2
+    const areas: {[key: string]: {
+        tabBarStyle: React.CSSProperties,
+        divStyle: React.CSSProperties
+    }} = {
+        'north': {
+            tabBarStyle: { left: 0, top: 0, width: width, height: tabBarHeight },
+            divStyle: {left: 0, top: tabBarHeight, width: width, height: H0}
+        },
+        'south': {
+            tabBarStyle: { left: 0, top: tabBarHeight + H0, width: width, height: tabBarHeight },
+            divStyle: { left: 0, top: tabBarHeight * 2 + H0, width: width, height: H0 }
+        }
+    }
+
+    const handleClick = useCallback((e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
```

Comment:
On mouse click anywhere within the north, assign active area to north. Similar for south.